lib/tasks/errors.rake

```diff
@@ -0,0 +1,41 @@
+namespace :errors do
+  desc <<-END_DESC
+Fetches log for given request id
+
+This task is useful to get log for given request id from logs in log directory. The request_id is required.
+You can also specify log file where request logs has been saved. Default option is 'production.log'.
+
+Available attributes:
+  * request_id = Request id from 500 page
+  * log_file = specify another log file to lookup (default is 'production.log')
+
+Examples:
+  # foreman-rake errors:fetch_log request_id=aabbccss5 [log_file=development.log]
+  END_DESC
+
+  task :fetch_log do
+    request_id = ENV['request_id']
+    logfile = ENV['log_file'] || 'production.log' # default file where we want to find is production.log
+    unless request_id
+      puts "Can't find log without request_id"
+      exit(1)
+    end
+    file_path = File.join(Foreman::Logging.log_directory, logfile)
+    unless File.exist?(file_path)
+      puts "Can't find log file #{file_path}"
+      exit(1)
+    end
+    result = `grep #{request_id} #{file_path}`
+    if result.empty?
+      puts "Can't find log for request_id"
+      exit(1)
+    end
+    puts result
+    puts "\n"
+    puts "Foreman version: #{Foreman::VERSION}"
+    puts "Plugins: "
+    Gem::Specification.all_names.each do |plugin_name|
+      puts " - #{plugin_name}"
+    end
+  end
+end
```

Comment:
Do I miss anything or is this just a 41 lines of code that does `grep request_id /var/log/foreman/production.log`? ;-)

I am not strictly against it, but less code means less bugs...

Reply:
Well, not exactly. You can specify what file you can scan. It's a production.log by default.

Reply:
I see no real use for this rake task.

Reply:
it's just a helper task to get all of a specific request id's lines from the log with a bit of useful metadata instead of leaving it up to the admin to figure out what they need to do

Reply:
Although I don't think we need to maintain another helper just because of this as we can present simply `grep` and `rpm -q foreman` commands on the error page itself, I think it could be more useful if we also grep the UUID in the `proxy.log` when available. Lots of users have all-in-one box deployments and request id propagates into proxy too, so we could actually see the whole transactions.

Only then this rake task would make more sense to me, although I still think we should ask the user to run three commands in the error page instead of this.

Reply:
@lzap yes, it will be much easier to display information for user how grep command use or just show them a path to logfile. However that doesn't solve the problem that this PR tries to solve because the path to logs can be security issue for somebody. So little abstraction to avoid this is needed.

On the other hand, I like the idea to attach also log from proxy with same request ID. What do you think @tbrisker?

Reply:
We can consider trying to add proxy log if it's there in a follow-up pr if needed.